docs/overview.md

```diff
@@ -0,0 +1,127 @@
+# Overview
+
+## What is djinni?
+
+Djinni is a tool for generating cross-language type declarations and interface bindings. It's designed to connect C++ with either Java or Objective-C.
+
+Djinni can be used to interface cross-platform C++ library code with platform-specific Java and Objective-C on Android and iOS. Dropbox announced Djinni at CppCon 2014.
+
+<iframe width="560" height="315" src="https://www.youtube-nocookie.com/embed/ZcBtF-JWJhM" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+!!! tip "Djinni ❤️ Community!"
+
+    Dropbox ended maintenance of Djinni in March 2020. This fork wants to give Djinni a second life by transforming it to a community-driven project.
+
+    We are currently in the early stage of migration to the all new [cross-language-cpp](https://github.com/cross-language-cpp) organization. ([More Information...](contributing.md))
+
+## How the magic works
+
+To call from one programming language to another, usually a lot of gluecode is required.
+Djinni automatically generates this gluecode from an interface file.
+
+All you need to do is implement the interfaces generated by djinni in one language and use it from the other language. Djinni takes care of the heavy lifting: Converting data-types, managing pointers and managing memory.
+
+
+??? example "Example: Call C++ Code from Java or Objective-C"
+
+    This Djinni Interface defines a class that is implemented in C++ and can be called from Java or Objective-C:
+    ```
+    HelloWorld = interface +c {
+        static create(): HelloWorld;
+        fromCpp(): string
+    }
+    ```
+
+
+
+    === "Implemented in C++"
+
+        ``` cpp
+        #include "djinni/generated/HelloWorld.hpp"
+
+        class HelloWorldImpl : public HelloWorld {
+        public:
+            static std::shared_ptr<HelloWorld> create() {
+                return std::make_shared<HelloWorldImpl>();
+            }
+            std::string fromCpp() {
+                return "Hello From C++!";
+            }
+        }
+        ```
+
+    === "Called from Java"
+
+        ``` java
+        HelloWorld helloWorld = HelloWorld.create();
+        System.out.println(helloWorld.fromCpp());
+        // Hello from C++!
+        ```
+
+    === "or Objective-C"
+
+        ``` objc
+        HelloWorld *helloWorld = [HelloWorld create];
+        NSLog([helloWorld fromCpp]);
```

Comment:
Do you speak Objective-C? I hacked this together after googling a few minutes, this may be wrong :D

Reply:
That's fine, but I will get help from some objective-C people at work, just to be sure